donut/modules/courses/helpers.py

```diff
@@ -303,3 +303,25 @@ def get_user_scheduler_sections(username, year, term):
 
 is_duplicate_error = lambda e: \
     isinstance(e, IntegrityError) and e.args[0] == ER.DUP_ENTRY
+
+
+def get_notes(username, course):
+    user_id = get_user_id(username)
+    query = "SELECT notes FROM scheduler_notes WHERE user_id=%s AND course_id=%s"
+    with flask.g.pymysql_db.cursor() as cursor:
+        cursor.execute(query, (user_id, course))
+        res = cursor.fetchone()
+        if res:
+            return res['notes']
+        else:
+            return None
+
+
+def edit_notes(username, course, notes):
+    user_id = get_user_id(username)
+    query = """
+    INSERT INTO scheduler_notes (user_id, course_id, notes) VALUES (%s, %s, %s)
+    ON DUPLICATE KEY UPDATE notes=%s
```

Comment:
You can avoid passing `notes` twice by doing `ON DUPLICATE KEY UPDATE notes = VALUES(notes)`

Reply:
Doesn't VALUES(notes) = the old notes value?

Reply:
No, that's just `notes`. `VALUES(notes)` is the new value of `notes` you are trying to insert. We already do this in the directory update script: https://github.com/ASCIT/donut/blob/master/donut/modules/directory_search/utils/import_registrar.py#L88-L91